daml-lf/data/src/main/scala/com/digitalasset/daml/lf/VersionRange.scala

```diff
@@ -22,24 +20,21 @@ import com.daml.lf.transaction.VersionTimeline
 final case class VersionRange[V](
     min: V,
     max: V,
-)(implicit ev: VersionTimeline.SubVersion[V]) {
-  import VersionTimeline._
-  import VersionTimeline.Implicits._
+)(implicit ordering: Ordering[V]) {
 
-  def intersect(that: VersionRange[V])(
-      implicit ev: VersionTimeline.SubVersion[V]): VersionRange[V] =
+  def intersect(that: VersionRange[V]): VersionRange[V] =
     VersionRange(
-      min = maxVersion(this.min, that.min),
-      max = minVersion(this.max, that.max)
+      min = this.min max that.min,
+      max = this.max min that.max,
     )
 
   def isEmpty: Boolean =
-    max precedes min
+    ordering.lt(max, min)
```

Comment:
It seems somewhat inconsistent to not write `max < min` here like everywhere else. 😄

Reply:
dropped